### PR TITLE
doc: tools: add tools section and pages

### PR DESCRIPTION
**Pull request by greg-fer**
Added tools section and pages in documentation.
This adds readmies for chip-cert and spake2p to the doc pool.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
     "examples/thermostat/nxp/linux-se05x/README.md",
     "examples/common/m5stack-tft/repo",
     "docs/guides/README.md",
+    "tools/memory/memdf/README.md",
+    "tools/memory/platform/README.md",
```
[the remark this answers is not in the document]
Comment (gautesl) [reply docs/conf.py:36]: Should also add to the `external_content_contents` list
```py
    (MATTER_BASE, "src/tools/**/*.md"),
    (MATTER_BASE, "scripts/tools/**/*.md"),
```

**Review by gautesl — COMMENTED** [review]
This still leaves some errors with some links to example folders

**Comment by greg-fer** [comment]
@andrei-menzopol , I've moved the NXP guide you added from platforms/nxp/doc to docs/guides and added it to the doc pool at https://project-chip.github.io/connectedhomeip-doc/guides/index.html , so the page will be visible there once this is merged. Could you check everything is OK and sign off this change? Thanks!

**Comment by andrei-menzopol** [comment]
@greg-fer, looks good. Thank you.

**Review by LuDuda — APPROVED** [review]
Thank you Greg!

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**